app/routes.py

```diff
@@ -95,3 +95,47 @@ def get_id(kml_admin_id):
         ),
         200
     )
+
+
+@app.route('/kml/<kml_admin_id>', methods=['PUT'])
+@validate_content_type("application/vnd.google-earth.kml+xml")
+def put_kml(kml_admin_id):
+    enforcer = DynamoDBFilesHandler(
+        table_name=AWS_DB_TABLE_NAME,
+        bucket_name=AWS_S3_BUCKET_NAME,
+        table_region=AWS_DB_REGION_NAME,
+        endpoint_url=AWS_DB_ENDPOINT_URL
+    )
+    item = enforcer.get_item(kml_admin_id)
+
+    # Fetching a non existing Item will return "None"
+    if item is None:
+        logger.error("Could not find the following kml id in the database: %s", kml_admin_id)
+        abort(400, f"Could not find {kml_admin_id} within the database.")
+
+    kml_id = item['file_id']
+
+    quoted_str = request.get_data().decode('utf-8')
+    data = unquote_plus(quoted_str)
+    kml_string = validate_kml_string(data)
+
+    executor = S3FileHandling(AWS_S3_REGION_NAME, AWS_S3_ENDPOINT_URL)
+    executor.upload_object_to_bucket(kml_id, kml_string, bucket_name=AWS_S3_BUCKET_NAME)
+
+    timestamp = time.strftime('%Y-%m-%d %X', time.localtime())
+    enforcer.update_item_timestamp(kml_admin_id, timestamp)
+
+    return make_response(
+        jsonify(
+            {
+                'success': True,
+                'id': kml_admin_id,
+                'links':
+                    {
+                        'self': f'{request.host_url}/kml/{item["admin_id"]}',
```

Comment:
With PR https://github.com/geoadmin/service-kml/pull/12 the `request.host_url` was introduced. This already contains a trailing `/`, so that this line should be:
`'self': f'{request.host_url}kml/{item["admin_id"]}',`
instead of
'self': f'{request.host_url}**/**kml/{item["admin_id"]}',

Otherwise there'd be double // between the `request.host_url` and `kml/`.
(Needs to be replaced in the responses of all routes that use `request.host_url`)